### PR TITLE
[CHERI-MIPS] Stop modelling all integer loads/stores as using $ddc

### PR DESCRIPTION
**Pull request by arichardson**
This creates huge diffs to upstream in all MIR tests and it seems unlikely
that modelling DDC accesses could improve code generation much. If code
is changing DDC, CWriteHwr acts as a hasSideEffects barrier and therefore
loads/stores should not be reordered before/after the CWriteHwr.

Mostly reverts commit c54e8413298908118018cee3270bbc838a200bb6

Fixes: https://github.com/CTSRD-CHERI/llvm-project/issues/570

### CODE REVIEW

**Comment by jrtc27** [comment]
Ah, C0 explains the origin of it... had C0-as-DDC been kept the instructions should just have been split up with GPCR-but-not-DDC and DDC versions.

File: llvm/lib/Target/Mips/MipsSEFrameLowering.cpp [hunk 1]
```diff
@@ -425,9 +425,9 @@ void MipsSEFrameLowering::emitPrologue(MachineFunction &MF,
   unsigned ADDiu = ABI.GetPtrAddiuOp();
   unsigned AND = ABI.IsN64() ? Mips::AND64 : Mips::AND;
 
-  // C0 is always live in.  In non-CHERI MIPS, C0 is treated as a read-only
-  // register that is implicitly read by loads and stores.
-  MBB.addLiveIn(Mips::DDC);
+  // DDC is always live in.
+  if (STI.isCheri())
+    MBB.addLiveIn(Mips::DDC);
```
Comment (jrtc27) [inline llvm/lib/Target/Mips/MipsSEFrameLowering.cpp:430]: Not needed now there's no modelling of it, this isn't done for CHERI-RISC-V

Comment (arichardson) [reply llvm/lib/Target/Mips/MipsSEFrameLowering.cpp]: Fixed in the new version.

**Review by jrtc27 — APPROVED** [review]
🥳

**Pull request merged**